### PR TITLE
services/horizon: Add more verbose logging for horizon start up

### PR DESCRIPTION
**Pull request by erika-sdf**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Add more verbose logging to horizon start up. Include lines that signal db migration and captive core set up, which should be logged to stdout since it happens before logging is set up.

### Why

It would make debugging easier.

### CODE REVIEW

**Review by jacekn — APPROVED** [review]
Thank you, it's a very nice improvement. It looks good to me but it would be good to get 2nd review from somebody with more go experience.

File: services/horizon/internal/flags.go [hunk 1]
```diff
@@ -509,6 +509,8 @@ func NewAppFromFlags(config *Config, flags support.ConfigOptions) (*App, error)
 	if config.Ingest && !config.EnableCaptiveCoreIngestion && config.StellarCoreDatabaseURL == "" {
 		return nil, fmt.Errorf("flag --%s cannot be empty", StellarCoreDBURLFlagName)
 	}
+
+	log.Infof("Initializing horizon...")
```
Comment (sreuland) [inline services/horizon/internal/flags.go:513]: does this output print on console, wasn't sure on significance of using this or stdLog, the desired outcome is to be on console correct, assume they both output there, was just asking for my own edification on logging preferences.

**Review by sreuland — APPROVED** [review]

File: services/horizon/internal/flags.go (hunk 1 above)
Comment (erika-sdf) [reply services/horizon/internal/flags.go:513]: Both `log` and `stdLog` calls print to console. The code that runs prior to the logger initialization uses the `stdLog` calls. I think  @jacekn 's feature request was specifically about seeing failures prior to the initialization, so the `stdLog` statements provide some context on what's happening.

File: services/horizon/internal/init.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ import (
 )
 
 func mustNewDBSession(subservice db.Subservice, databaseURL string, maxIdle, maxOpen int, registry *prometheus.Registry) db.SessionInterface {
+	log.Infof("Establishing database session at %s for %v", databaseURL, subservice)
```
Comment (bartekn) [inline services/horizon/internal/init.go:21]: Sorry for a late review. I was OOO. Not sure if we should print `databaseURL` as it contains a sensitive info (password).

**Pull request merged**